File: Easy/145.cpp
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root){
        vector<int> res;
        solve(root, res);
        return res;
    }
    vector<int> solve(TreeNode* root, vector<int>&ans) {
        if(root == nullptr) return ans;
        solve(root->left, ans);
        solve(root->right, ans);
        ans->push_back(root->val);
        return ans;
    }
};

File: Medium/Tree/145.py
# postorder traverse binary tree

from typing import List

class TreeNode:
    def __init__(self,val):
        self.val = val
        self.left = None
        self.right = None

class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        res = []
        def dfs(node):
            if not node:
                return
            dfs(node.left)
            dfs(node.right)
            res.append(node.val)

        dfs(root)
        return res

